File: 2020-01/29/chris.java
import java.util.*;

class Solution {
    public List<String> fizzBuzz(final int n) {
        final List<String> results = new ArrayList<>(n);

        for (int i = 1; i <= n; i++) {
            results.add(this.getResult(i));
        }

        return results;
    }

    private String getResult(final int n) {
        if (n % 15 == 0) {
            return "FizzBuzz";
        }

        if (n % 3 == 0) {
            return "Fizz";
        }

        if (n % 5 == 0) {
            return "Buzz";
        }

        return String.valueOf(n);
    }
}

File: 2020/2020-01/29/chris.rs
impl Solution {
    pub fn fizz_buzz(n: i32) -> Vec<String> {
        let mut vec: Vec<String> = Vec::new();

        for i in 1..(n + 1) {
            vec.push(get_result(i));
        }

        return vec;
    }
}

fn get_result(number: i32) -> String {
    if number % 15 == 0 {
        return String::from("FizzBuzz");
    }

    if number % 3 == 0 {
        return String::from("Fizz");
    }

    if number % 5 == 0 {
        return String::from("Buzz");
    }

    return number.to_string();
}
